phoenix-hive/src/main/java/org/apache/phoenix/hive/util/PhoenixUtil.java

```diff
@@ -0,0 +1,163 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.hive.util;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+
+public class PhoenixUtil {
+    static Log LOG = LogFactory.getLog(PhoenixUtil.class.getName());
+
+    public static boolean createTable(Connection conn, String TableName,
+            Map<String, String> fields, String[] pks, boolean addIfNotExists, int salt_buckets,
+            String compression,int versions_num) throws SQLException, MetaException {
+        Preconditions.checkNotNull(conn);
+        if (pks == null || pks.length == 0) {
+            throw new SQLException("Phoenix Table no Rowkeys specified in "
+                    + HiveConfigurationUtil.PHOENIX_ROWKEYS);
+        }
+        for (String pk : pks) {
+            String val = fields.get(pk.toLowerCase());
+            if (val == null) {
+                throw new MetaException("Phoenix Table rowkey " + pk
+                        + " does not belong to listed fields ");
+            }
+            val += " not null";
+            fields.put(pk, val);
+        }
+
+        StringBuffer query = new StringBuffer("CREATE TABLE ");
+        if (addIfNotExists) {
+            query.append("IF NOT EXISTS ");
+        }
+        query.append(TableName + " ( ");
+        Joiner.MapJoiner mapJoiner = Joiner.on(',').withKeyValueSeparator(" ");
+        query.append(" " + mapJoiner.join(fields));
+        if (pks != null && pks.length > 0) {
+            query.append(" CONSTRAINT pk PRIMARY KEY (");
+            Joiner joiner = Joiner.on(" , ");
+            query.append(" " + joiner.join(pks) + " )");
+        }
+        query.append(" )");
+        if (salt_buckets > 0) {
+            query.append(" SALT_BUCKETS = " + salt_buckets);
+        }
+        if (compression != null) {
+            query.append(" ,COMPRESSION='GZ'");
```

Comment:
Is there a reason that GZ is the default compression? Why not just let Phoenix default it? Same for VERSIONS below.